<?xml version="1.0"?>
<robot xmlns:xacro="https://www.ros.org/wiki/xacro">


    <ros2_control name="HardwareSystem" type="system">
        <hardware>
            <plugin>curio_one/CurioBotSystemHardware</plugin>
            <param name="left_wheel_name">left_wheel_joint</param>
            <param name="right_wheel_name">right_wheel_joint</param>
            <param name="device">/dev/ttyACM0</param>
            <param name="baud_rate">57600</param>
            <param name="timeout_ms">1000</param>
            <param name="loop_rate">30</param>
            <param name="enc_counts_per_rev">752</param>
        </hardware>
        <joint name="left_wheel_joint">
            <command_interface name="velocity" />
            <state_interface name="position" />
            <state_interface name="velocity" />
        </joint>

        <joint name="right_wheel_joint">
            <command_interface name="velocity" />
            <state_interface name="position" />
            <state_interface name="velocity" />
        </joint>

    </ros2_control>


</robot>